stdlib/@python2/itertools.pyi

```diff
@@ -22,8 +22,14 @@ class chain(Iterator[_T], Generic[_T]):
 
 def compress(data: Iterable[_T], selectors: Iterable[Any]) -> Iterator[_T]: ...
 def dropwhile(predicate: Callable[[_T], Any], iterable: Iterable[_T]) -> Iterator[_T]: ...
-def ifilter(predicate: Callable[[_T], Any] | None, iterable: Iterable[_T]) -> Iterator[_T]: ...
-def ifilterfalse(predicate: Callable[[_T], Any] | None, iterable: Iterable[_T]) -> Iterator[_T]: ...
+@overload
+def ifilter(predicate: None, iterable: Iterable[_T | None]) -> Iterator[_T]: ...
+@overload
+def ifilter(predicate: Callable[[_T], Any], iterable: Iterable[_T]) -> Iterator[_T]: ...
+@overload
+def ifilterfalse(predicate: None, iterable: Iterable[_T | None]) -> Iterator[_T]: ...
```

Comment:
I don't think this is correct. Using Python 3:

```python
>>> from itertools import filterfalse
>>> list(filterfalse(None, [0,1,2,3,None]))
[0, None]
```

I don't think an overload for `ifilterfalse` makes sense here.